### PR TITLE
Fix broken range requests

### LINKED ISSUE #481: Proxy range requests - Test range request failed
image

This may be related to recent changes in AWS SDK https://github.com/catalyst/moodle-local_aws/pull/36

### PR DESCRIPTION
**Pull request by golenkovm**
This is to fix #481

### CODE REVIEW

File: classes/check/proxy_range_request.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        $url = new \moodle_url('/admin/category.php', ['category' => 'tool_objectfs']);
+        return new \action_link($url, get_string('pluginname', 'tool_objectfs'));
+    }
+
+    /**
+     * Check for the success of a proxied range request, if the setting is enabled.
+     *
+     * @return result
+     */
+    public function get_result() : result {
```
Comment (keevan) [inline classes/check/proxy_range_request.php:45]:
```suggestion
    public function get_result(): result {
```

File: lib.php [hunk 2]
```diff
@@ -110,3 +110,18 @@ function tool_objectfs_pluginfile($course, $cm, context $context, $filearea, arr
...
+ * @return array
+ */
+function tool_objectfs_status_checks() {
+    if (get_config('tool_objectfs', 'proxyrangerequests')) {
+        return [
+            new tool_objectfs\check\proxy_range_request()
+        ];
+    } else {
+        return [];
+    }
```
Comment (keevan) [inline lib.php:126]:
```suggestion
    return [];
```

File: classes/check/proxy_range_request.php (hunk 1 above)
Comment (Peterburnett) [reply classes/check/proxy_range_request.php]: Thanks @keevan , fixed

**Comment by golenkovm** [comment]
Cool, it's looking good @Peterburnett 
By the way, should we also have a check for objectfs connection status?

**Review by keevan — APPROVED** [review]

**Pull request merged**